### PR TITLE
Docs for set_filetype command

### LINKED ISSUE #5084: Add docs for `set_filetype`
As reported in #5047, the `set_filetype` command is missing any type of documentation.

### PR DESCRIPTION
**Pull request by unho**
Fixes #5084.

### CODE REVIEW

File: docs/server/commands.rst [hunk 1]
```diff
@@ -419,6 +419,49 @@ Available options:
...
+^^^^^^^^^^^^
+
+.. versionadded:: 2.8
+
+This command sets file formats for projects, and also allows to convert files
+to another format.
+
+.. django-admin-option:: --from-filetype
+
+Specifies to convert stores of this file type.
```
Comment (dwaynebailey) [inline docs/server/commands.rst]: Convert stores of this file type.

File: docs/server/commands.rst [hunk 2]
```diff
@@ -419,6 +419,49 @@ Available options:
...
+
+.. django-admin-option:: --matching
+
+Specifies which stores to convert by matching their path within the project,
+excluding file extension.
```
Comment (dwaynebailey) [inline docs/server/commands.rst]: Convert stores matching this path within the project.

File: docs/server/commands.rst [hunk 3]
```diff
@@ -419,6 +419,49 @@ Available options:
...
+
+`--from-filetype` and `--matching` can be both specified at the same time.
```
Comment (dwaynebailey) [inline docs/server/commands.rst]: I'm not sure this is needed in the docs, I'd only make a commnt if they were mutually exclusive, I'd take this for granted.

File: docs/server/commands.rst [hunk 4]
```diff
@@ -419,6 +419,49 @@ Available options:
...
+.. code-block:: console
+
+    $ pootle set_filetype --project=myproj --from-filetype=po properties
+
+
+For example, to add a new format to a project, run:
+
+.. code-block:: console
+
+    $ pootle set_filetype --project=myproj --matching=mydir/myfile-* properties
```
Comment (dwaynebailey) [inline docs/server/commands.rst]: I'm not 100% sure I understand `--matching` based on this.  I'd assume I'd use `--matching` to convert certain files to a new filetype.  I.e. why didn't I just use your first example to achieve what you've specified here?

Comment (unho) [inline docs/server/commands.rst]: First example and this one achieve different things. The only way they would behave the same is if in the project all files that don't have the specified target filetype (`properties`) are exactly the files in the project that match the given `--matching` glob.

Comment (unho) [inline docs/server/commands.rst]: I will rephrase the comment anyway.

**Pull request merged**